### ISSUE TITLE
[Bug]: sp-slider only adds one sp-slider-handle when multiple are defined

### ISSUE THREAD

**Issue by peterjanes** [adobe/spectrum-web-components#2035]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-slider, sp-slider-handle

### Expected behavior

Adding an `sp-slider` with multiple handles from an HTML `template` should add all of the handles.

### Actual behavior

I haven't nailed down the exact conditions, but sometimes when adding `sp-slider`s dynamically from an HTML `template` only one instance of `sp-slider-handle` will be added because the handle map key is `undefined`. (I finally tracked down the behaviour when I started getting "First slider handle cannot have attribute min='previous'" after adding `max="next"` to the first `sp-slider-handle` and `min="previous"` to the second.)

### What browsers are you seeing the problem in?

Firefox, Chrome

### How can we reproduce this issue?

I haven't been able to come up with a reduced test case, but the issue is 100% reproducible in my (currently non-public) app and seems to be due to some instances of a field not being renamed. (`handle.name` is used instead of `handle.handleName` in `HandleController.ts`.) I'll create an associated pull request for review.

**Comment by Westbrook** [comment]
@peterjanes thanks for sharing this issue!

Before we can fully review changes in support of what you're experiencing, we will need a reproduction case so that we can compare your findings with ours and ensure that all users of this element are being appropriately served. I'd suggest you start from something like https://webcomponents.dev/edit/collection/fO75441E1Q5ZlI0e9pgq/U7LQv7LsAVBwJayJXG3B/src/index.ts and see if you can isolate the issue you are experiencing.

**Comment by peterjanes** [comment]
Yeah, I've been working on a reproduce case for a couple of days but can't get it to fail. Figured I'd submit the PR that fixes it (or appears to) as it seems like a pretty obvious update; `handle.name` is only set when `handle.handleName` isn't set (and only used in a log message), and the `undefined` `handle.handleName` is immediately used as the map key.

**Comment by peterjanes** [comment]
Here's my current non-failing reproduce case, which mirrors what the live code does:

```html
<!DOCTYPE html>
<script src="https://jspm.dev/@spectrum-web-components/bundle/elements.js" type="module"></script>

<div>
<sp-theme scale="medium" color="light">
</sp-theme>
</div>

<template id="rangeTemplate">
  <sp-slider variant="range" step="1">
    <sp-slider-handle slot="handle" name="min" label="Minimum" max="next"></sp-slider-handle>
    <sp-slider-handle slot="handle" name="max" label="Maximum" min="previous"></sp-slider-handle>
  </sp-slider>
</template>

<script module>
async function foo() {
  const res = await fetch('.');
  const rangeGroups = new Map();
  rangeGroups.set('group1', [{id: 'range1.1', min: 10, max: 90}, {id: 'range1.2', min: 15, max: 85}]);
  rangeGroups.set('group2', [{id: 'range2.1', min: 20, max: 80}, {id: 'range2.2', min: 25, max: 75}]);

  const rangesList = document.createElement('div');
  const rangeTemplate = document.querySelector('#rangeTemplate');

  rangeGroups.forEach((rangeGroup, key) => {
    const rangeList = document.createElement('div');
    rangeGroup.forEach((range) => {
      const rangeElement = rangeTemplate.content.cloneNode(true);
      const slider = rangeElement.querySelector('sp-slider');
      slider.setAttribute('id', range.id);
      slider.setAttribute('label', range.id);
      slider.setAttribute('min', String(range.min));
      slider.setAttribute('max', String(range.max));
      slider.querySelector('[name=min]').setAttribute('value', String(range.min));
      slider.querySelector('[name=max]').setAttribute('value', String(range.max));
      rangeList.appendChild(rangeElement);
    });
    rangesList.appendChild(rangeList);
  });
  document.querySelector('sp-theme').appendChild(rangesList);
  return 'done';
}
foo().then(console.log);
</script>
```

**Comment by Westbrook** [comment]
I assume you're not using `<script src="https://jspm.dev/@spectrum-web-components/bundle/elements.js" type="module"></script>` in your production code. Is there anything about how you're importing the elements that you can share, along with which versions you're leveraging?

**Comment by peterjanes** [comment]
In the live app I've reduced that whole thing to

```js
function createElement(html) {
  const temp = document.createElement('template');
  temp.innerHTML = html.trim();
  return temp.content.firstChild;
}

async function foo() {
  await fetch('.');
  document.body.appendChild(
      createElement(`<div>
  <sp-slider variant="range" step="1" id="price" name="price" label="Max Price" min="35425" max="86610">
    <sp-slider-handle slot="handle" name="min" label="Minimum" max="next" value="35425"></sp-slider-handle>
    <sp-slider-handle slot="handle" name="max" label="Maximum" min="previous" value="86610"></sp-slider-handle>
  </sp-slider>
</div>`);
}

foo();
```

and it still fails.

The app loads the dependencies with

```
import '@spectrum-web-components/theme/theme-light';
import '@spectrum-web-components/theme/scale-medium';
import '@spectrum-web-components/theme/sp-theme';
import '@spectrum-web-components/button/sp-button';
import '@spectrum-web-components/slider/sp-slider';
import '@spectrum-web-components/slider/sp-slider-handle';
```

from

```
    "@spectrum-web-components/button": "^0.16.1",
    "@spectrum-web-components/slider": "^0.12.1",
    "@spectrum-web-components/theme": "^0.9.1",
```

**Comment by Westbrook** [comment]
How is JS loaded into the app? For instance [this](https://codepen.io/Westbrook/pen/XWeBPjm?editors=1000) (working demo from above) uses `<script type="module">` which works different that a `<script>`, a dynamic import, etc.

**Comment by peterjanes** [comment]
```
<script type="module" src="script.js"></script>
```

**Comment by peterjanes** [comment]
I'm going to try again to strip down the app to something I can share instead of trying to build up a test case. Will let you know.

**Comment by peterjanes** [comment]
https://home.peterjanes.ca/~peterj/2022/bug-2035/ has a fairly minimal failing test case, with no recompilation, subsetting, etc. of the Spectrum components.

**Comment by peterjanes** [comment]
The fix appears to resolve #2032 as well.
